### PR TITLE
Created template for referencing Azure Storage secrets from ARM

### PR DESCRIPTION
**Pull request by EldertGrootenboer**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: arm-template-retrieve-azure-storage-access-keys/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+                "accessTier": "Hot",
+                "supportsHttpsTrafficOnly": true
+            }
+        },
+        {
+            "type": "Microsoft.Logic/workflows",
+            "apiVersion": "2017-07-01",
+            "name": "[parameters('logicAppName')]",
+            "location": "[parameters('location')]",
+            "dependsOn": ["[resourceId('Microsoft.Web/connections', parameters('storageConnectionName'))]"],
```
Comment (bmoore-msft) [inline arm-template-retrieve-azure-storage-access-keys/azuredeploy.json]: format the json (e.g. SHIFT+ALT+F in VS Code)

File: arm-template-retrieve-azure-storage-access-keys/azuredeploy.parameters.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+      "storageAccountName": {
+        "value": "GEN-UNIQUE"
+      },
+      "storageAccountSku": {
```
Comment (bmoore-msft) [inline arm-template-retrieve-azure-storage-access-keys/azuredeploy.parameters.json]: remove the parameters you have defaultValues for... so deployment can test your defaultValues (Location you have is fine if you want to deploy there)

File: arm-template-retrieve-azure-storage-access-keys/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            "type": "Microsoft.Web/connections",
+            "apiVersion": "2018-07-01-preview",
+            "name": "[parameters('storageConnectionName')]",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName'))]"
+            ],
+            "properties": {
+                "api": {
+                    "id": "[concat(subscription().id,'/providers/Microsoft.Web/locations/', parameters('location'), '/managedApis/azureblob')]"
```
Comment (bmoore-msft) [inline arm-template-retrieve-azure-storage-access-keys/azuredeploy.json]: remove concat and use resourceId

resourceId('Microsoft.Web/locations/managedApis', parameters('location'), 'azureblob')

File: arm-template-retrieve-azure-storage-access-keys/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+                        }
+                    }
+                },
+                "parameters": {
+                    "$connections": {
+                        "value": {
+                            "azureblob": {
+                                "connectionId": "[resourceId('Microsoft.Web/connections', parameters('storageConnectionName'))]",
+                                "connectionName": "azureblob",
+                                "id": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Web/locations/', parameters('location'), '/managedApis/azureblob')]"
```
Comment (bmoore-msft) [inline arm-template-retrieve-azure-storage-access-keys/azuredeploy.json]: remove concat - use resourceId per the above example.

File: arm-template-retrieve-azure-storage-access-keys/azuredeploy.json (hunk 3 above)
Comment (EldertGrootenboer) [reply arm-template-retrieve-azure-storage-access-keys/azuredeploy.json]: This doesn't work for the managed APIs.
The concat is also how this is described in the official documentation, f.e. https://docs.microsoft.com/en-us/azure/logic-apps/logic-apps-azure-resource-manager-templates-overview#connection-resource-definitions.
Also this is the same in other Logic App templates in the quickstart samples.

File: arm-template-retrieve-azure-storage-access-keys/azuredeploy.json (hunk 4 above)
Comment (EldertGrootenboer) [reply arm-template-retrieve-azure-storage-access-keys/azuredeploy.json]: See above reply.

**Comment by bmoore-msft** [comment]
I made the fixes for a subscriptionResourceId

**Pull request merged**